sql/catalyst/src/main/antlr3/org/apache/spark/sql/catalyst/parser/SparkSqlParser.g

```diff
@@ -2320,6 +2320,19 @@ regularBody[boolean topLevel]
    )
    |
    selectStatement[topLevel]
+   |
+   (LPAREN selectStatement[true]) => nestedSetOpSelectStatement[topLevel]
+   ;
+
+nestedSetOpSelectStatement[boolean topLevel]
+   :
+   (
+   LPAREN s=selectStatement[topLevel] RPAREN -> {$s.tree}
+   )
+   (set=setOpSelectStatement[$nestedSetOpSelectStatement.tree, topLevel])
```

Comment:
I've tried few approaches unsuccessfully. This works and I think it can also make recursively nested UNION query work. I will try if there is more simple approach tomorrow.